### PR TITLE
fix(Earthfile): Add .exe extension in the windows binaries

### PR DESCRIPTION
**Pull request by tampakrap**
### Description of your changes

This is a regression compared to the Makefile era, see for example
https://releases.crossplane.io/stable/v1.16.0/bin/windows_amd64/ and
also what install.sh expects.

I have: 

- [X] Read and followed Crossplane's [contribution process].
- [X] Run `earthly +reviewable` to ensure this PR is ready for review.
- [ ] ~Added or updated unit tests.~
- [ ] ~Added or updated e2e tests.~
- [ ] ~Linked a PR or a [docs tracking issue] to [document this change].~
- [ ] ~Added `backport release-x.y` labels to auto-backport this PR.~

Need help with this checklist? See the [cheat sheet].

[contribution process]: https://github.com/crossplane/crossplane/tree/master/contributing
[docs tracking issue]: https://github.com/crossplane/docs/issues/new
[document this change]: https://docs.crossplane.io/contribute/contribute
[cheat sheet]: https://github.com/crossplane/crossplane/tree/master/contributing#checklist-cheat-sheet

### CODE REVIEW

File: Earthfile [hunk 1]
```diff
@@ -152,12 +152,16 @@ go-build:
   ARG GOFLAGS="-ldflags=-X=github.com/crossplane/crossplane/internal/version.version=${CROSSPLANE_VERSION}"
   ARG CGO_ENABLED=0
   FROM +go-modules
+  LET extension = ""
```
Comment (jbw976) [inline Earthfile:155]: super nit: what do you think about calling this `ext` just to make it shorter?

**Review by jbw976 — COMMENTED** [review]
This looks like a reasonable approach to me @tampakrap. I'd like to test it a bit though, depending on the outcome of the conversation in https://github.com/crossplane/crossplane/pull/5779#pullrequestreview-2100502766 😎

File: Earthfile (hunk 1 above)
Comment (tampakrap) [reply Earthfile]: sure, done, I was juggling between naming it "extensions" or "ext" actually 😸

**Comment by jbw976** [comment]
With https://github.com/crossplane/crossplane/pull/5782 now merged, @tampakrap do you want to fetch/rebase on later in master, then we can review testing and reviewing this PR? 🙏

**Comment by jbw976** [comment]
Also, the DCO check looks to be failing with the latest commit, so hopefully you can ammed/clean that up too!

https://github.com/crossplane/crossplane/tree/master/contributing#certificate-of-origin

**Comment by tampakrap** [comment]
> With #5782 now merged, @tampakrap do you want to fetch/rebase on later in master, then we can review testing and reviewing this PR? 🙏

sure, rebased and squashed the commits into one

**Review by jbw976 — APPROVED** [review]
This looks good to me @tampakrap! From my testing on my Mac M1, first starting with an empty `_output` dir:

Run a windows go-build:
```
> earthly --platform=windows/amd64 +go-build
```

We have just the windows binaries with the `.exe` extension:
```
> tree _output
_output
└── bin
    └── windows_amd64
        ├── crank.exe
        └── crossplane.exe

2 directories, 2 files
```

Then also checking to make sure that the regular build didn't regress:
```
❯ earthly +build
```

We now also have the Mac binaries, without the `.exe` extension:
```
❯ tree _output
_output
├── bin
│   ├── darwin_arm64
│   │   ├── crank
│   │   └── crossplane
│   └── windows_amd64
│       ├── crank.exe
│       └── crossplane.exe
└── charts
    └── crossplane-0.0.0-1718052420-bc5cfbd4.tgz

4 directories, 5 files
```

**Review by negz — APPROVED** [review]

**Comment by negz** [comment]
Thank you, good catch!

**Pull request merged**